src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,108 @@
+import React, {
+  useState,
+  useMemo,
+  useCallback,
+  useRef,
+} from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import { PeoplesNames } from '../types/Person';
+
+type Props = {
+  peoplesNames: PeoplesNames[];
+  onSelected: (person: string) => void;
+  delay: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  peoplesNames,
+  onSelected,
+  delay,
+}) => {
+  const [inputQuery, setInputQuery] = useState('');
+  const [processedQuery, setProcessedQuery] = useState('');
+  const [suggestions, setSuggestions] = useState<PeoplesNames[]>([]);
+  const [isFocused, setIsFocused] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const handleDelayedChange = useCallback(
+    debounce(setProcessedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputQuery(event.target.value);
+    handleDelayedChange(event.target.value);
+  };
+
+  useMemo(() => {
```

Comment:
You `useMemo` like `useEffect` it doesn't make sense.
[Docs](https://react.dev/reference/react/useMemo)

If you want to use useMemo here, it should look like this, and then render only `filteredSuggestions`

```javascript
const filteredSuggestions = useMemo(
    () => peoplesNames
      .filter((person) => person.name.toLowerCase().includes(processedQuery.toLowerCase())),
    [processedQuery]
  );
```